capture/listener_test.go

```diff
@@ -439,30 +431,34 @@ func testChunkedSequence(t *testing.T, listener *Listener, packets ...*TCPPacket
 	if len(listener.respAliases) != 0 {
 		t.Fatal("respAliases non empty:", listener.respAliases)
 	}
-
-	if len(listener.respWithoutReq) != 0 {
-		t.Fatal("respWithoutReq non empty:", listener.respWithoutReq)
-	}
 }
 
-func permutation(n int, list []*TCPPacket) []*TCPPacket {
-	if len(list) == 1 {
-		return list
+// permutation using heap algorithm https://en.wikipedia.org/wiki/Heap%27s_algorithm
+func permutation(a []*TCPPacket, f func([]*TCPPacket)) {
+	n := len(a)
+	c := make([]int, n)
+	f(a)
+	i := 0
+	for i < n {
+		if c[i] < i {
+			if i&1 != 1 {
+				a[0], a[i] = a[i], a[0]
+			} else {
+				a[c[i]], a[i] = a[i], a[c[i]]
+			}
+			f(a)
+			c[i]++
+			i = 0
+		} else {
+			c[i] = 0
+			i++
+		}
 	}
-
-	k := n % len(list)
-
-	first := []*TCPPacket{list[k]}
-	next := make([]*TCPPacket, len(list)-1)
-
-	copy(next, append(list[:k], list[k+1:]...))
-
-	return append(first, permutation(n/len(list), next)...)
 }
 
 // Response comes before Request
 func TestRawListenerChunkedWrongOrder(t *testing.T) {
-	listener := NewListener("", "0", EnginePcap, true, 10*time.Millisecond, "", "", 0, false, false)
+	listener := NewListener("", "0", EnginePcap, true, 10*time.Second, "", "", 0, false, false)
```

Comment:
Why did you change expire to 10 sec?

Reply:
The parsing of packets was taking longer than `1o milliseconds` but it won't spend `10s` its to give it enough time and the logic of the tests will not be affected